src/redisearch_rs/rqe_iterators/src/rqe_iterator.rs

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use ffi::t_docId;
+use inverted_index::RSIndexResult;
+
+/// The result of an iterator read or skipping operation
+pub enum RQEIteratorStatus {
+    /// The iterator has found a valid result, in either `read()` or `skip_to()`
+    OK(RSIndexResult),
+
+    /// A `skip_to()` has found a result with a `doc_id` greater than the requested `doc_id`
+    NotFound(RSIndexResult),
+}
+
+/// An iterator failure indications
+pub enum RQEIteratorError {
+    /// The iterator has reached the time limit for execution.
+    TimedOut,
+}
+
+/// The status of the iterator after a call to `revalidate`
+pub enum RQEValidateStatus {
+    /// The iterator is still valid and at the same position.
+    OK,
```

Comment:
```suggestion
    Ok,
```

Enum variant names should be camelcased.